buildkite/README.md

```diff
@@ -251,6 +251,28 @@ platforms:
 In this example the jobs on Windows and MacOS would use 0.20.0, whereas the job on Ubuntu would run 0.18.0.
 Please see the [Bazelisk documentation](https://github.com/philwo/bazelisk/blob/master/README.md#how-does-bazelisk-know-which-version-to-run) for a list of all supported version values.
 
+### macOS: Using a specific version of Xcode
+
+We upgrade the CI machines to the latest version of Xcode shortly after it is released and this
+version will then be used as the default Xcode version. If required, you can specify a fixed Xcode
+version to test against in your pipeline config.
+
+```yaml
+tasks:
+  macos_xcode_10_1:
+    platform: macos
+    xcode_version: "10.1"
+    build_targets:
+    - "..."
+  macos_xcode_10_2:
+    platform: macos
+    xcode_version: "10.2"
+    build_targets:
+    - "..."
+```
```

Comment:
what's the policy going to be going forward? internally, we use the latest 2 minor versions, plus the last available version of the previous major version.

for example, if this is what would be available internally:

* 10.2.1 <--
* 10.2.0
* 10.1.2 <--
* 10.1.1
* 10.1.0
* 10.0.1
* 10.0.0
* 9.4.1 <--
* 9.4.0
* 9.3.1
* 9.3.0

This simplifies the support needed to have multiple versions of Xcode.

One thing I'd like to point out, is to mention this Xcode version support specifically as a solution to avoid being broken because of latest Xcodes, but that the general policy is to _not_ specify an Xcode version at all so that the default can be upgraded by your team more easily over time. If all clients start specifying the Xcode version, then it will be more work for your CI team to migrate to newer Xcode versions, as you'll also need to update client configs.

Reply:
Good question. Let's say that the policy is that we won't remove Xcode versions for now, until it becomes a problem to do so. Recently Apple doesn't seem to release patch-level versions of Xcode, but if they start doing so, we could just adopt your scheme.

Do you think it's reasonable to just automatically transform e.g. "10.2" in a config file to "10.2.1" in case Apple releases that patch level version? Then we could simplify the regex to only allow MAJOR.MINOR and just always pick the latest patch-level version for that.

Reply:
that depends on whether you'll have 10.2.0 and 10.2.1 at the same time. I'd recommend not having them at the same time, and when 10.2.1 comes out, replace 10.2.0 with the new one. Otherwise, specifying 10.2 is going to be confusing since there are 2 versions that match.

Reply:
Yes, my idea would be that users just don't care about the patch-level version in their configs. They just say "xcode_version: 10.2" and that just maps to the latest patch-level version installed on the machine, whether that's 10.2, 10.2.1 or 10.2.2. The CI log would also show the exact version including the patch level.

Yeah, I would only keep the latest patch-level version of each MAJOR.MINOR pair.

Does that make sense?

Reply:
SGTM